main.rb

```diff
@@ -38,75 +42,90 @@ def my_select
     array
   end
 
-  def my_all?(_argv = 0)
+  def my_all?(argv = nil)
     i = 0
-    j = 0
     array = []
     while i < size
       if block_given?
         return false unless yield(self[i])
-
-        j += 1
-        array.push(self[i])
-        return true if j == size
       else
-        return false unless self[i]
-
-        j += 1
+        if argv.is_a? Class
+          return true if is_a? argv
+        elsif argv.is_a? Regexp
+          return false unless self[i].match(argv)
+        elsif argv.nil?
+          return false unless self[i]
+        end
         array.push(self[i])
-        return true if j == size
       end
       i += 1
     end
-    return true if array.empty?
+    return false unless array.empty?
+
+    true
   end
 
-  def my_any?(_argv = 0)
+  def my_any?(argv = nil)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words = %w[dog door rod blade]
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true
```